src/main/java/com/ldtteam/structurize/network/messages/ClientBlueprintRequestMessage.java

```diff
@@ -0,0 +1,122 @@
+package com.ldtteam.structurize.network.messages;
+
+import com.ldtteam.structurize.Network;
+import com.ldtteam.structurize.api.util.Log;
+import com.ldtteam.structurize.storage.StructurePacks;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.util.Tuple;
+import net.minecraft.world.level.block.Mirror;
+import net.minecraft.world.level.block.Rotation;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.network.NetworkEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * Request a blueprint from the client.
+ */
+public class ClientBlueprintRequestMessage implements IMessage
+{
+    /**
+     * List of pending blueprint requests.
+     */
+    public static final Queue<Tuple<ClientBlueprintRequestMessage, Future<byte[]>>> blueprintRequestQueue = new LinkedList<>();
```

Comment:
No. This is not part of the message, move this to some dedicated mechanism or handler.